### PR TITLE
Pub 1841

### PR DESCRIPTION
**Pull request by NatashaAlker**
### JIRA link

https://tools.hmcts.net/jira/browse/PUB-1841

### Change description

Functional tests for Admin Dashboard User Management

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/test/end-to-end/tests/admin/user-management-test.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    I.waitForText('User Updated');
+
+    I.click('Home');
+    I.deleteAccount(TEST_EMAIL);
+});
+
+Scenario('I as an admin user should be able to see all errors related to modify a user account', async ({ I }) => {
+    I.loginAsAdmin();
+    I.waitForText('Your Dashboard');
+    I.createAdminAccount(TEST_FIRST_NAME, TEST_LAST_NAME, TEST_EMAIL, TEST_ROLE);
```
Comment (KianKwa) [inline src/test/end-to-end/tests/admin/user-management-test.ts:64]: What is the purpose of creating the account and deleting here? I can see this account getting used in this scenario

File: src/test/end-to-end/tests/admin/user-management-test.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    I.click('Continue');
+    I.waitForText('Manage ' + TEST_EMAIL);
+    I.click('Change');
+    I.waitForText('What role would you like ' + TEST_EMAIL + ' to have?');
+    I.selectOption('updatedRole', 'CTSC Admin');
+    I.click('Continue');
+    I.waitForText('User Updated');
+
+    I.click('Home');
+    I.deleteAccount(TEST_EMAIL);
```
Comment (KianKwa) [inline src/test/end-to-end/tests/admin/user-management-test.ts:58]: We also need to test the delete account functionality is working as intended. May be we should go back to the main  screen after account deletion and check that we get the error message "There is no user matching that email address" if we try to search for TEST_EMAIL

File: src/test/end-to-end/tests/admin/user-management-test.ts (hunk 1 above)
Comment (NatashaAlker) [reply src/test/end-to-end/tests/admin/user-management-test.ts:64]: The scenario is to alter the role for the account and I was asked not to perform this on an existing account :)

File: src/test/end-to-end/tests/admin/user-management-test.ts (hunk 2 above)
Comment (NatashaAlker) [reply src/test/end-to-end/tests/admin/user-management-test.ts:58]: I've added this check for completeness :)

**Review by KianKwa — DISMISSED** [review]

File: src/test/end-to-end/tests/admin/user-management-test.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+Feature('Manage admin user accounts');
+
+const TEST_WORKER_NUMBER = randomData.getRandomNumber(10000000, 99999999);
+const TEST_FIRST_NAME = testConfig.TEST_SUITE_PREFIX + 'FirstName';
+const TEST_LAST_NAME = testConfig.TEST_SUITE_PREFIX + 'Surname';
+const TEST_EMAIL = randomData.getRandomEmailAddress(TEST_WORKER_NUMBER);
+const TEST_ROLE = 'Internal - Administrator - Local';
+const TEST_INVALID_EMAIL = 'invalid_email@test.com';
+
```
Comment (ashwini-mv) [inline src/test/end-to-end/tests/admin/user-management-test.ts:12]: All test email addresses should end up with either justice,gov.uk or hmcts.net, even the invalid addresses :)

**Review by ashwini-mv — CHANGES_REQUESTED** [review]

**Review by KianKwa — APPROVED** [review]

**Review by ashwini-mv — APPROVED** [review]

**Pull request merged**